regex-automata/src/util/determinize/state.rs

```diff
@@ -460,12 +460,10 @@ impl<'a> Repr<'a> {
     ///
     /// If this state is not a match state, then this always returns 0.
     fn match_len(&self) -> usize {
-        if !self.is_match() {
-            return 0;
-        } else if !self.has_pattern_ids() {
-            1
-        } else {
-            self.encoded_pattern_len()
+        match (self.is_match(), self.has_pattern_ids()) {
+            (false, _) => 0,
+            (true, false) => 1,
+            (true, true) => self.encoded_pattern_len(),
```

Comment:
I find the pre-existing code easier to read. Although it looks like you could at least change `return 0;` to `0`.